packages/app/src/cli/services/app/config/push.ts

```diff
@@ -0,0 +1,35 @@
+import {selectApp} from '../select-app.js'
+import {PushConfig, PushConfigSchema} from '../../../api/graphql/push_config.js'
+import {parseConfigurationFile} from '../../../models/app/loader.js'
+import {AppConfigurationSchema, AppInterface} from '../../../models/app/app.js'
+import {ensureAuthenticatedPartners} from '@shopify/cli-kit/node/session'
+import {partnersRequest} from '@shopify/cli-kit/node/api/partners'
+import {AbortError} from '@shopify/cli-kit/node/error'
+import {renderSuccess} from '@shopify/cli-kit/node/ui'
+import {OutputMessage} from '@shopify/cli-kit/node/output'
+
+export interface Options {
+  apiKey: string
+  app: AppInterface
+}
+
+export async function pushConfig(options: Options) {
+  const token = await ensureAuthenticatedPartners()
+  const apiKey = options.apiKey || (await selectApp()).apiKey
+  const mutation = PushConfig
+
+  const configuration = await parseConfigurationFile(AppConfigurationSchema, options.app.configurationPath, abort)
+  const variables = {apiKey, ...configuration}
+  const result: PushConfigSchema = await partnersRequest(mutation, token, variables)
+
+  if (result.appUpdate.userErrors.length > 0) {
+    const errors = result.appUpdate.userErrors.map((error) => error.message).join(', ')
```

Comment:
We should also include which field was invalid in the error message. This is what I got with a very long app url:
![screenshot with error message without field name](https://github.com/Shopify/cli/assets/446046/ef591d08-5578-482b-b5ee-02c78bbb0bb1)

Reply:
is it cool to do this in a followup? i'd like to merge this to get @RyanDJLee unblocked